src/client/Microsoft.Identity.Client/Cache/CacheInfo.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.;
+
+namespace Microsoft.Identity.Client.Cache
+{
+    /// <summary>
+    /// Enum describing the reason for cache refresh and fetching access token from ESTS.
+    /// </summary>
+    public enum CacheInfo
```

Comment:
Since we already have the cache info with the refresh in value in it from telemetry. Does it make sense to expose this object. Perhaps this other information can be useful at some point

Reply:
Also, the name CacheInfoTelemetry didnt seem right if this is to be exposed. What are your thoughts on this change?

Should I only expose the refresh in value?

@bgavrilMS

Reply:
We have a specific request now from @rymeskar to expose this, so definitely let's do it.

Reply:
From a naming perspective, I think this is not intuitive. This enum tells the user the reason why MSAL had to contact AAD to get a token. So maybe the name should be more like `CacheMissReason` ?

Reply:
Not sure if `CacheMiss` is accurate for all values. I think `CacheMiss` applies to `NoCachedAT` and `Expired`. `None` and `ForceRefresh` don't even check the cache and `ForceRefresh` returns a valid token.

> This enum tells the user the reason why MSAL had to contact AAD to get a token.

This describes it well but I'm not sure of a more accurate class name.

Reply:
Maybe `CacheUpdateReason`, `CacheUpdateHint` or `CacheUpdateIndication`
@pmaytak @bgavrilMS?

Reply:
`TokenSourceReason`? We have `TokenSource` already (cache, IdP, Broker), and this enum will explain why it went to that source?

Reply:
The only problem with `TokenSourceReason` is it wont make sense when the `TokenSource` value is `Cache`. For example, if you have a `TokenSource` of `Cache` and the `TokenSourceReason` is `ProactivelyRefreshed`, it doesnt explain why the token came from the cache. However, it does explain why the token cache was updated.

This is why I suggested `CacheUpdateReason` as it is more general and makes sense with all of the enums

Reply:
The only problem with `TokenSourceReason` is it wont make sense when the `TokenSource` value is `Cache`. For example, if you have a `TokenSource` of `Cache` and the `TokenSourceReason` is `ProactivelyRefreshed`, it doesnt explain why the token came from the cache. However, it does explain why the token cache was refrshed.

I suggest we do `CacheRefreshReason` as it is more general and makes sense with all of the enums

Reply:
> The only problem with `TokenSourceReason` is it wont make sense when the `TokenSource` value is `Cache`. For example, if you have a `TokenSource` of `Cache` and the `TokenSourceReason` is `ProactivelyRefreshed`, it doesnt explain why the token came from the cache. However, it does explain why the token cache was refrshed.
> 
> I suggest we do `CacheRefreshReason` as it is more general and makes sense with all of the enums
Yea, makes sense.